WIP: Initial policy for installkernel, dracut updates

0xC0ncord opened the pull request:
Initial policy for installkernel and lots of dracut updates.

cgzones committed:
ci: add module storage test

Add test to verify modular policies can be successfully installed from
a module storage.

Signed-off-by: Christian Göttsche <cgzones@googlemail.com>
Signed-off-by: Chris PeBenito <pebenito@ieee.org>

pebenito committed:
build-userspace/setools.yml: Add commit hash to cache ID.

The jobs use the git reference to cheeckout the sources.  Additional
commits may have added since the cache was created, so add the commit ID to
the cache's key.

Signed-off-by: Chris PeBenito <pebenito@ieee.org>

0xC0ncord committed:
installkernel: initial policy

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
installkernel: initial policy

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
dracut: initial import from Gentoo

Upstreaming current version of the dracut policy from Gentoo.

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
dracut: initial import from Gentoo

Upstreaming current version of the dracut policy from Gentoo.

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
dracut: cleanup

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
dracut: cleanup

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
dracut: various fixes

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
dracut: various fixes

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
testing: allow dracut_t to use CAP_SYS_ADMIN

dracut uses fsfreeze when building an initrd.

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
testing: allow dracut_t to use CAP_SYS_ADMIN

dracut uses fsfreeze when building an initrd.

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
contrib: drop dracut

WavyEbuilder commented:
Just a note: probably want to add a closes line to the commit for https://bugs.gentoo.org/704980

0xC0ncord committed:
installkernel: add filetrans for /var/lib/misc/installkernel

Signed-off-by: Kenton Groombridge <me@concord.sh>

WavyEbuilder committed:
xserver: fixup labelling for mesa cache

Label mesa_shader_cache_db[1][2], mesa_shader_cache_sf[3][4] and
radv_builtin_shaders[5] as mesa_shader_cache_t.

[1] https://gitlab.freedesktop.org/mesa/mesa/-/blob/5d92c92ce56c4c3b9d98936fee51129281923e1a/docs/envvars.rst#L263-266
[2] https://gitlab.freedesktop.org/mesa/mesa/-/blob/5d92c92ce56c4c3b9d98936fee51129281923e1a/src/util/disk_cache.h#L48
[3] https://gitlab.freedesktop.org/mesa/mesa/-/blob/5d92c92ce56c4c3b9d98936fee51129281923e1a/docs/envvars.rst#L219-232
[4] https://gitlab.freedesktop.org/mesa/mesa/-/blob/5d92c92ce56c4c3b9d98936fee51129281923e1a/src/util/disk_cache.h#L47
[5] https://gitlab.freedesktop.org/mesa/mesa/-/blob/5d92c92ce56c4c3b9d98936fee51129281923e1a/src/amd/vulkan/radv_physical_device.c#L2495

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>

WavyEbuilder committed:
wayland: allow reading udev runtime files for input devices

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>

WavyEbuilder committed:
wayland: grant access to /dev/input for libinput

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>

WavyEbuilder committed:
wayland: allow managing wayland_runtime_t files for display lock file

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>

WavyEbuilder committed:
wayland: allow talking to logind over dbus

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>

WavyEbuilder committed:
wayland: allow managing wayland_runtime_t sock files for display sock

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>

WavyEbuilder committed:
wayland: grant wayland_compositor access to /dev/udmabuf

For linux-dmabuf-v1: https://wayland.app/protocols/linux-dmabuf-v1

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>

WavyEbuilder committed:
wayland: allow violating W^X for compositors behind a boolean

This is probably to do with llvmpipe. Further analysis to come later.

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>

WavyEbuilder committed:
chromium: drop the chromium_render_t domain

Upstream no longer supports the SELinux sandbox[1], so no dyntrans is
attempted, making this domain redundent. It's been long enough now that
it is reasonable to drop this domain.

[1] https://github.com/chromium/chromium/blob/0f8100c82dc431809d866e5d0218cbe3c482b20c/sandbox/linux/README.md?plain=1#L31

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>

0xC0ncord committed:
init, systemd: add interfaces for systemd managed unit dirs

systemd supports automatically managing directories required by services
via RuntimeDirectory=, StateDirectory=, and other directives. When a
unit with these starts, init will create these directories before the
service itself is started. Of these, RuntimeDirectory= is the only one
which will also be deleted when the unit stops.

Add generic interfaces that allows a policy module with its own file
types to allow init to manage these.

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
tuned: various fixes

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
systemd: allow socket activation for systemd-resolved

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
systemd: dontaudit user session getattr generic blk devices

Seen on a system with NBD devices from Ceph. For some reason
systemd --user instances spuriously try to getattr on these devices,
generating a lot of noise.

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
netutils: allow network access in a user namespace

You can attach to another network namespace with nsenter. If the
namespace happens to also be associated to a user namespace, the access
against cap_userns is checked.

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

dsugar100 committed:
fapolicyd: support for new /usr/sbin/fapolicyd-rpm-loader

avc:  denied  { setattr } for  pid=1141 comm="fapolicyd" name="fapolicyd" dev="tmpfs" ino=1306 scontext=system_u:system_r:fapolicyd_t:s0 tcontext=system_u:object_r:var_run_t:s0 tclass=dir permissive=0

avc:  denied  { execute } for  pid=5631 comm="fapolicyd" name="fapolicyd-rpm-loader" dev="dm-0" ino=174938 scontext=system_u:system_r:fapolicyd_t:s0 tcontext=system_u:object_r:bin_t:s0 tclass=file permissive=1
avc:  denied { execute_no_trans } for  pid=5631 comm="fapolicyd" path="/usr/sbin/fapolicyd-rpm-loader" dev="dm-0" ino=174938 scontext=system_u:system_r:fapolicyd_t:s0 tcontext=system_u:object_r:bin_t:s0 tclass=file permissive=1

avc:  denied  { read } for  pid=5630 comm="fapolicyd" path="pipe:[157491]" dev="pipefs" ino=157491 scontext=system_u:system_r:fapolicyd_t:s0 tcontext=system_u:system_r:fapolicyd_t:s0 tclass=fifo_file permissive=1
avc:  denied { getattr } for  pid=5631 comm="fapolicyd-rpm-l" path="pipe:[157491]" dev="pipefs" ino=157491 scontext=system_u:system_r:fapolicyd_t:s0 tcontext=system_u:system_r:fapolicyd_t:s0 tclass=fifo_file permissive=1
avc:  denied  { write } for  pid=5631 comm="fapolicyd-rpm-l" path="pipe:[157491]" dev="pipefs" ino=157491 scontext=system_u:system_r:fapolicyd_t:s0 tcontext=system_u:system_r:fapolicyd_t:s0 tclass=fifo_file permissive=1

Signed-off-by: Dave Sugar <dsugar100@gmail.com>

etbe committed:
Label ~/.cache/gstreamer-[0-9\.]+(/.*)? files (#1042)

Signed-off-by: Russell Coker <russell@coker.com.au>

etbe committed:
games (#1026)

* Some small changes for games_t

Signed-off-by: Russell Coker <russell@coker.com.au>

etbe committed:
Wireshark patch to allow execmem (which it unfortunately needs), allow (#1039)

netlink_generic_socket access, allow reading kernel network state and crypro
sysctls, access DRI, and have exec access to libraries.

Signed-off-by: Russell Coker <russell@coker.com.au>

dsugar100 committed:
firewalld: Allow firewall-cmd to be called from systemd

RedHat has added 'ExecStartPost=/usr/bin/firewall-cmd --state' in the firewalld.service file.

node=localhost type=USER_AVC msg=audit(1763093230.354:10021): pid=1156 uid=81 auid=4294967295 ses=4294967295 subj=system_u:system_r:system_dbusd_t:s0 msg='avc:  denied  { send_msg } for  scontext=system_u:system_r:initrc_t:s0 tcontext=system_u:system_r:firewalld_t:s0 tclass=dbus permissive=1 exe="/usr/bin/dbus-broker" sauid=81 hostname=? addr=? terminal=?' UID="dbus" AUID="unset" SAUID="dbus"

Signed-off-by: Dave Sugar <dsugar100@gmail.com>

0xC0ncord committed:
Merge upstream

WavyEbuilder committed:
gcc_config_t: allow reading cgroup files for cpu.max

Various coreutils, such as sort, now attempt[1] to read cpu.max due to
a change in gnulib[2]:

> avc:  denied  { read } for  pid=22163 comm="sort" name="cpu.max" dev="cgroup2" ino=1808 scontext=superuser:sysadm_r:gcc_config_t tcontext=system_u:object_r:cgroup_t tclass=file permissive=0

[1] https://gitweb.git.savannah.gnu.org/gitweb/?p=coreutils.git;a=commit;h=c4df55be7f78d29415abd7765f863ff38184f86e
[2] https://gitweb.git.savannah.gnu.org/gitweb/?p=gnulib.git;a=commit;h=9b07115f4a344effef1dde8bd0e6e356d4b0e744

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>

GargiQcom committed:
Add sepolicy for bootloader to create directory in dosfs

type=AVC msg=audit(1754407901.439:65): avc:  denied  { create } for pid=625 comm="bootctl" name="loader" scontext=system_u:system_r:bootloader_t:s0 tcontext=system_u:object_r:dosfs_t:s0 tclass=dir permissive=0

Signed-off-by: Gargi Misra <gmisra@qti.qualcomm.com>

GargiQcom committed:
Adding rules for mount to search configfs

type=AVC msg=audit(1754407902.111:87): avc:  denied  { search } for  pid=724 comm="mount" name="/" dev="configfs" ino=5121 scontext=system_u:system_r:mount_t:s0 tcontext=system_u:object_r:configfs_t:s0 tclass=dir permissive=0
type=AVC msg=audit(1754407902.111:88): avc:  denied  { search } for  pid=724 comm="mount" name="adb" dev="configfs" ino=14873 scontext=system_u:system_r:mount_t:s0 tcontext=system_u:object_r:configfs_t:s0 tclass=dir permissive=0

Signed-off-by: Gargi Misra <gmisra@qti.qualcomm.com>

GargiQcom committed:
Adding rules for virt module

type=AVC msg=audit(1754407902.331:103): avc:  denied  { search } for  pid=738 comm="virtlockd" name="sbin" dev="sda2" ino=11213 scontext=system_u:system_r:virtlockd_t:s0 tcontext=system_u:object_r:bin_t:s0 tclass=dir permissive=0
type=AVC msg=audit(1754407902.343:105): avc:  denied  { search } for  pid=738 comm="virtlockd" name="ssl" dev="sda2" ino=389 scontext=system_u:system_r:virtlockd_t:s0 tcontext=system_u:object_r:cert_t:s0 tclass=dir permissive=0
type=AVC msg=audit(1754407903.219:106): avc:  denied  { read } for  pid=764 comm="virtlockd" name="openssl.cnf" dev="sda2" ino=692 scontext=system_u:system_r:virtlockd_t:s0 tcontext=system_u:object_r:cert_t:s0 tclass=file permissive=0

Signed-off-by: Gargi Misra <gmisra@qti.qualcomm.com>

21sw-clayton committed:
logging: allow syslogd_t syslog_tls_port_t name_connect

rsyslogd[492]: cannot connect to example.home.arpa:6514: Permission
denied [v8.2402.0 try https://www.rsyslog.com/e/2027 ]

--

type=PROCTITLE proctitle=/usr/sbin/rsyslogd -n -iNONE

type=SOCKADDR saddr={ saddr_fam=inet laddr=1.2.3.4 lport=6514 }

type=SYSCALL arch=aarch64 syscall=connect success=no
exit=EACCES(Permission denied) a0=0x6 a1=0x7fff38060bc0 a2=0x10 a3=0x0
items=0 ppid=1 pid=492 auid=unset uid=root gid=root euid=root suid=root
fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=unset
comm=rs:main Q:Reg exe=/usr/sbin/rsyslogd
subj=system_u:system_r:syslogd_t:s0 key=(null)

type=AVC avc:  denied  { name_connect } for  pid=492 comm=rs:main Q:Reg
dest=6514 scontext=system_u:system_r:syslogd_t:s0
tcontext=system_u:object_r:syslog_tls_port_t:s0 tclass=tcp_socket

--

Fedora:

https://github.com/fedora-selinux/selinux-policy/commit/42504eb364b73234bd622fe674427bdfb68dc043

$ sesearch -A --source syslogd_t --target syslog_tls_port_t --perm name_connect
allow syslogd_t syslog_tls_port_t:tcp_socket { name_bind name_connect };

Signed-off-by: Clayton Casciato <ccasciato@21sw.us>

21sw-clayton committed:
logging: allow miscfiles_read_generic_certs(syslogd_t)

rsyslog needs to be able to access certificates to validate encrypted
communications.

https://docs.redhat.com/de/documentation/red_hat_enterprise_linux/10/html/risk_reduction_and_recovery_operations/configuring-a-remote-logging-solution#configuring-tls-encrypted-remote-logging

--

rsyslogd: error: defaultnetstreamdrivercafile file
'/etc/pki/ca-trust/source/anchors/ca-cert.pem' could not be accessed:
Permission denied [v8.2402.0 try https://www.rsyslog.com/e/2039 ]

--

type=PROCTITLE proctitle=/usr/sbin/rsyslogd -n -iNONE

type=SYSCALL arch=aarch64 syscall=openat success=no
exit=EACCES(Permission denied) a0=AT_FDCWD a1=0x5555d6951ce0
a2=O_RDONLY|O_CLOEXEC a3=0x0 items=0 ppid=1 pid=258 auid=unset uid=root
gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root
tty=(none) ses=unset comm=rs:main Q:Reg exe=/usr/sbin/rsyslogd
subj=system_u:system_r:syslogd_t:s0 key=(null)

type=AVC avc:  denied  { search } for  pid=258 comm=rs:main Q:Reg
name=pki dev="vda" ino=234 scontext=system_u:system_r:syslogd_t:s0
tcontext=system_u:object_r:cert_t:s0 tclass=dir permissive=0

--

Fedora:

https://github.com/fedora-selinux/selinux-policy/commit/66742ce1af1d752fbefe8a512f16c9165634d07f

Signed-off-by: Clayton Casciato <ccasciato@21sw.us>

0xC0ncord committed:
container: allow execute_no_trans on container_tmpfs_t files

Recent alpine containers have apk triggers that perform an
execute_no_trans on tmpfs files. This needs to be explicitly allowed.

----
type=PROCTITLE msg=audit(12/23/2025 11:02:01.450:437892) : proctitle=apk add clang lld perl make
type=SYSCALL msg=audit(12/23/2025 11:02:01.450:437892) : arch=x86_64 syscall=execve success=no exit=EACCES(Permission denied) a0=0x78ce2303f039 a1=0x6628e6bafe88 a2=0x6628e6baff48 a3=0x0 items=0 ppid=1123910 pid=1124034 auid=unset uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=unset comm=apk exe=/sbin/apk subj=system_u:system_r:container_t:s0:c136,c227 key=(null)
type=AVC msg=audit(12/23/2025 11:02:01.450:437892) : avc:  denied  { execute_no_trans } for  pid=1124034 comm=apk path=/memfd:lib/apk/exec/busybox-1.37.0-r30.trigger (deleted) dev="tmpfs" ino=716831 scontext=system_u:system_r:container_t:s0:c136,c227 tcontext=system_u:object_r:container_tmpfs_t:s0:c136,c227 tclass=file permissive=0
----

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

WavyEbuilder commented on policy/modules/admin/dracut.te:
```diff
@@ -0,0 +1,207 @@
+	systemd_read_user_units_files(dracut_t)
+')
+
+ifdef(`distro_gentoo',`
```
This should probably be optional, all our other modules (e.g. puppet) seem to prefer optional for portage blocks over ifdef, which makes sense to me (as theoretically too gentoo supports using a different PMS implementation).

GargiQcom committed:
Suppressing denial for systemd login to read process init scripts avc: denied { search } for pid=736 comm="systemd-logind" name="944" dev="proc" ino=15804 scontext=system_u:system_r:systemd_logind_t:s0 tcontext=system_u:system_r:initrc_t:s0 tclass=dir permissive=0

Signed-off-by: Gargi Misra <gmisra@qti.qualcomm.com>

pebenito committed:
init: Fix typo in init.if comment

Signed-off-by: Chris PeBenito <pebenito@ieee.org>

pebenito committed:
tests.yml: Increase userspace to v3.9.

Signed-off-by: Chris PeBenito <pebenito@ieee.org>

dsugar100 committed:
snmpd doesn't seem to need sys_ptrace capability

Tested on RHEL9 with net-snmp 5.9.4.
I was able to query information about running processes.  I'm not sure
what else sys_ptrace might be needed for by snmpd

Signed-off-by: Dave Sugar <dsugar100@gmail.com>

pebenito committed:
ci: Create artifact of the workdir if the build fails.

Signed-off-by: Chris PeBenito <pebenito@ieee.org>

pebenito committed:
validate/diff-policy.yml: Fix library usage.

It was using the Ubuntu distro libs instead of the built ones.

Signed-off-by: Chris PeBenito <pebenito@ieee.org>

21sw-clayton committed:
authlogin: label /var/lib/lastlog (lastlog2)

https://github.com/fedora-selinux/selinux-policy/commit/bbbaa364c2b93795127a1ffbf0f90b244753d75a
"Label /var/lib/lastlog with lastlog_t"

Signed-off-by: Clayton Casciato <ccasciato@21sw.us>

21sw-clayton committed:
authlogin: add auth_create_lastlog and auth_delete_lastlog

Signed-off-by: Clayton Casciato <ccasciato@21sw.us>

21sw-clayton committed:
locallogin: allow local_login_t lastlog_t create,delete

Note that the { read } AVC denial was not addressed.

It is currently unknown what specifically triggers this and Fedora
policy does not appear to allow this.

--

Fedora:

https://github.com/fedora-selinux/selinux-policy/commit/fe29879463b7176dab24c0a9210131fa6e7cd130
"Allow systemd (PID 1) create lastlog entries"

--

pam_lastlog2(login:session): Cannot open database
(/var/lib/lastlog/lastlog2.db): unable to open database file

PROCTITLE proctitle=2F62696E2F6C6F67696E002D70002D2D00726F6F74

AVC avc:  denied  { getattr } for  pid=244 comm="login"
path="/var/lib/lastlog" dev="vda" ino=32776
scontext=system_u:system_r:local_login_t:s0
tcontext=system_u:object_r:lastlog_t:s0 tclass=dir permissive=1

SYSCALL arch=c00000b7 syscall=79 success=yes exit=0 a0=ffffffffffffff9c
a1=55557ab4db98 a2=7ffff6dd39c0 a3=100 items=0 ppid=1 pid=244
auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0
tty=ttyAMA0 ses=4294967295 comm="login" exe="/usr/bin/login.shadow"
subj=system_u:system_r:local_login_t:s0 key=(null)

--

PROCTITLE proctitle=2F62696E2F6C6F67696E002D70002D2D00726F6F74

AVC avc:  denied  { search } for  pid=244 comm="login" name="lastlog"
dev="vda" ino=32776 scontext=system_u:system_r:local_login_t:s0
tcontext=system_u:object_r:lastlog_t:s0 tclass=dir permissive=1

SYSCALL arch=c00000b7 syscall=79 success=no exit=-2 a0=ffffffffffffff9c
a1=55557ab4db98 a2=7ffff6dd39c0 a3=100 items=0 ppid=1 pid=244
auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0
tty=ttyAMA0 ses=4294967295 comm="login" exe="/usr/bin/login.shadow"
subj=system_u:system_r:local_login_t:s0 key=(null)

--

PROCTITLE proctitle=2F62696E2F6C6F67696E002D70002D2D00726F6F74

AVC avc:  denied  { write } for  pid=244 comm="login" name="lastlog"
dev="vda" ino=32776 scontext=system_u:system_r:local_login_t:s0
tcontext=system_u:object_r:lastlog_t:s0 tclass=dir permissive=1

AVC avc:  denied  { add_name } for  pid=244 comm="login"
name="lastlog2.db" scontext=system_u:system_r:local_login_t:s0
tcontext=system_u:object_r:lastlog_t:s0 tclass=dir permissive=1

AVC avc:  denied  { create } for  pid=244 comm="login"
name="lastlog2.db" scontext=system_u:system_r:local_login_t:s0
tcontext=system_u:object_r:lastlog_t:s0 tclass=file permissive=1

SYSCALL arch=c00000b7 syscall=56 success=yes exit=4 a0=ffffffffffffff9c
a1=55557ab4e2b4 a2=88042 a3=1a4 items=4 ppid=1 pid=244 auid=4294967295
uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=ttyAMA0
ses=4294967295 comm="login" exe="/usr/bin/login.shadow"
subj=system_u:system_r:local_login_t:s0 key=(null)

CWD cwd="/"

PATH item=0 name=(null) inode=32776 dev=fe:00 mode=040755 ouid=0 ogid=0
rdev=00:00 obj=system_u:object_r:lastlog_t:s0 nametype=PARENT cap_fp=0
cap_fi=0 cap_fe=0 cap_fver=0 cap_frootid=0

PATH item=1 name=(null) nametype=CREATE cap_fp=0 cap_fi=0 cap_fe=0
cap_fver=0 cap_frootid=0

PATH item=2 name=(null) inode=32776 dev=fe:00 mode=040755 ouid=0 ogid=0
rdev=00:00 obj=system_u:object_r:lastlog_t:s0 nametype=PARENT cap_fp=0
cap_fi=0 cap_fe=0 cap_fver=0 cap_frootid=0

PATH item=3 name=(null) inode=32867 dev=fe:00 mode=0100644 ouid=0 ogid=0
rdev=00:00 obj=system_u:object_r:lastlog_t:s0 nametype=CREATE cap_fp=0
cap_fi=0 cap_fe=0 cap_fver=0 cap_frootid=0

--

PROCTITLE proctitle=2F62696E2F6C6F67696E002D70002D2D00726F6F74

AVC avc:  denied  { read } for  pid=244 comm="login" name="lastlog"
dev="vda" ino=32776 scontext=system_u:system_r:local_login_t:s0
tcontext=system_u:object_r:lastlog_t:s0 tclass=dir permissive=1

AVC avc:  denied  { open } for  pid=244 comm="login"
path="/var/lib/lastlog" dev="vda" ino=32776
scontext=system_u:system_r:local_login_t:s0
tcontext=system_u:object_r:lastlog_t:s0 tclass=dir permissive=1

SYSCALL arch=c00000b7 syscall=56 success=yes exit=6 a0=ffffffffffffff9c
a1=7ffff6dd5740 a2=80000 a3=0 items=0 ppid=1 pid=244 auid=4294967295
uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=ttyAMA0
ses=4294967295 comm="login" exe="/usr/bin/login.shadow"
subj=system_u:system_r:local_login_t:s0 key=(null)

--

PROCTITLE proctitle=2F62696E2F6C6F67696E002D70002D2D00726F6F74

AVC avc:  denied  { remove_name } for  pid=244 comm="login"
name="lastlog2.db-journal" dev="vda" ino=32868
scontext=system_u:system_r:local_login_t:s0
tcontext=system_u:object_r:lastlog_t:s0 tclass=dir permissive=1

AVC avc:  denied  { unlink } for  pid=244 comm="login"
name="lastlog2.db-journal" dev="vda" ino=32868
scontext=system_u:system_r:local_login_t:s0
tcontext=system_u:object_r:lastlog_t:s0 tclass=file permissive=1

SYSCALL arch=c00000b7 syscall=35 success=yes exit=0 a0=ffffffffffffff9c
a1=55557ab4e2d2 a2=0 a3=7fff4a1abb68 items=0 ppid=1 pid=244
auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0
tty=ttyAMA0 ses=4294967295 comm="login" exe="/usr/bin/login.shadow"
subj=system_u:system_r:local_login_t:s0 key=(null)

Signed-off-by: Clayton Casciato <ccasciato@21sw.us>

pebenito committed:
build-policy: Add temp backwards compatibility

For transition time to new validation target

Signed-off-by: Chris PeBenito <pebenito@ieee.org>

yizhao1 committed:
systemd: allow journalctl to search log directory

Allow journalctl to search log directory:

avc: denied { read } for pid=483 comm="journalctl" name="log"
dev="vda" ino=64466 scontext=system_u:system_r:systemd_journal_init_t:s0
tcontext=system_u:object_r:var_log_t:s0 tclass=lnk_file permissive=1

avc: denied { search } for pid=483 comm="journalctl" name="log"
dev="tmpfs" ino=2 scontext=system_u:system_r:systemd_journal_init_t:s0
tcontext=system_u:object_r:var_log_t:s0 tclass=dir permissive=1

Fix: #1060

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>

0xC0ncord committed:
podman: allow watching systemd-resolved runtime

aardvark-dns watches the resolv.conf files created by systemd-resolved.
Without this, DNS resolution fails.

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
init: allow using system container BPF programs

When a system container engine like CRI-O is used, it may require BPF
programs for managing container networking. Allow init to run these
programs.

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
crio: various fixes

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
kanidm: hookup systemd unit dir management

Signed-off-by: Kenton Groombridge <concord@gentoo.org>

0xC0ncord committed:
Merge upstream
